bin/node/src/service.rs

```diff
@@ -183,30 +186,36 @@ pub fn new_partial(config: &Configuration) -> Result<ServiceComponents, ServiceE
     })
 }
 
-#[allow(clippy::type_complexity)]
-#[allow(clippy::too_many_arguments)]
-fn setup(
-    config: Configuration,
-    backend: Arc<FullBackend>,
-    chain_status: SubstrateChainStatus,
-    keystore_container: &KeystoreContainer,
-    import_queue: sc_consensus::DefaultImportQueue<Block>,
-    transaction_pool: Arc<sc_transaction_pool::FullPool<Block, FullClient>>,
-    task_manager: &mut TaskManager,
-    client: Arc<FullClient>,
-    telemetry: &mut Option<Telemetry>,
-    import_justification_tx: mpsc::UnboundedSender<Justification>,
-    collect_extra_debugging_data: bool,
-) -> Result<
-    (
-        RpcHandlers,
-        SubstrateNetwork<Block, BlockHash>,
-        NetworkStarter,
-        SyncOracle,
-        Option<ValidatorAddressCache>,
-    ),
-    ServiceError,
-> {
+struct AlephRuntimeVars {
+    pub session_period: SessionPeriod,
+    pub millisecs_per_block: MillisecsPerBlock,
+}
+
+fn get_aleph_runtime_vars(client: &Arc<FullClient>) -> AlephRuntimeVars {
+    let finalized = client.info().finalized_hash;
+
+    let session_period = SessionPeriod(client.runtime_api().session_period(finalized).unwrap());
+
+    let millisecs_per_block =
+        MillisecsPerBlock(client.runtime_api().millisecs_per_block(finalized).unwrap());
```

Comment:
Here and above plx use `expect` with an explanation as to why (i.e. that we always have these, right?).

Reply:
done